autotech.lua

```diff
@@ -390,20 +391,79 @@ function auto_tech:set_tech_unit()
         error()
     end
 
+    --get the depths of all science packs on the tech tree, assuming non-progression military science
+    self.technology_nodes:for_all_nodes(function(technology_node)
+        local factorio_tech = technology_node.object_node.object
+        local science_pack_unlocked_by_this_tech = data.raw.tool[technology_node.object_node.object.name]
+        if science_pack_unlocked_by_this_tech then
+            if factorio_tech.name == "military-science-pack" then
```

Comment:
I see no reason not to merge the two logging checks and store the depth after logging. This is over-complicated as-is.